### PR TITLE
v0.7 upgrade

### PR DESCRIPTION
**Pull request by evetion**
### Added
- Improved documentation by using VitePress and showcasing all methods on a DEM.
- Package extensions on GeoArrays and Rasters that automatically set the `cellsize` for all methdos using that kwarg, taking into account geographical data (lat/lon).
- Added hydrology operators: `filldepressions`, `flowaccumulation`, with different algorithms (D8, DInf, FD8) available via the `method` kwarg.
- Added *multiscale* options to relative terrain operations using a `window` kwarg for a Stencil from Stencils.jl package.
- Added `BPI` as a relative terrain operation.
- Overhauled curvature methods by introducing `plan_curvature`, `profile_curvature` and `contour_curvature`, while deprecating `curvature` for `laplacian`.
- Added direction kwarg to slope and curvature methods.
- Added `radius` kwarg to curvature methods.
- Added this `CHANGELOG.md`

### Changed
- Using Stencils instead of LocalFilters.
- Refactored spread to choose from multiple algorithms with the `method` kwarg.

### Fixed
- Relaxed `Array` input to `AbstractArray` for `opening`

### CODE REVIEW

**Comment by evetion** [comment]
@alex-s-gardner For which curvatures (plan, profile, contour) do you want to use the direction argument? Do you have something working currently?

File: ext/GeomorphometryRastersExt.jl [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    (step(dim[1]) * degwidth * cosd(centercoords[2]), step(dim[2]) * degwidth)
+end
+
+function _crstrait(dem::Raster)
+    crs = Rasters.crs(dem)
+    acrs = ArchGDAL.importCRS(crs)
+    Bool(ArchGDAL.GDAL.osrisgeographic(acrs.ptr)) && return Rasters.GI.GeographicTrait()
+    Bool(ArchGDAL.GDAL.osrisprojected(acrs.ptr)) && return Rasters.GI.ProjectedTrait()
+    return Rasters.GI.UnknownTrait()
+end
```
Comment (evetion) [inline ext/GeomorphometryRastersExt.jl:36]: @rafaqz @tiemvanderdeure Ideally we implement crstrait on Rasters itself. And there is a cellarea method, but ideally we have the underlying cellsize (resolution?) as well. If we get a unit with that, even better.

**Comment by alex-s-gardner** [comment]
> @alex-s-gardner For which curvatures (plan, profile, contour) do you want to use the direction argument? Do you have something working currently?

I missed your earlier ping: I currently use directional profile curvature... instead of calculation curvature along maximum slope I calculate it along a direction.

**Comment by evetion** [comment]
> > @alex-s-gardner For which curvatures (plan, profile, contour) do you want to use the direction argument? Do you have something working currently?
> 
> I missed your earlier ping: I currently use directional profile curvature... instead of calculation curvature along maximum slope I calculate it along a direction.

Do you have a code example/snippet? My code is here: https://github.com/Deltares/Geomorphometry.jl/pull/41/files#diff-5e200c4ade1e6d95f1c218e8fa25406ee2b424c81f5dff53ecc59c1c641adc4fR295-R334, can be a bit hard to follow. A direction of 0 degrees is north. Might be best to just plot and compare, and see whether they're similar.

**Comment by evetion** [comment]
That's only the dz/dx and dz/dy calculation (so slope, not curvature?). Docs are now live at https://deltares.github.io/Geomorphometry.jl/dev/usage#curvature

**Comment by alex-s-gardner** [comment]
It's curvature as it subtracts the central value `v[5]` and divides by l[x/y] .^2  so its ddz/dx and ddy/dy

It's `D` and `E` in the first example found here: https://pro.arcgis.com/en/pro-app/latest/tool-reference/spatial-analyst/how-curvature-works.htm

But not worth spending any time on ... i use it to identify correlated biases in DEMs so I only use it as a relative metric of curvature... not something you should replicate... sorry for the confusion

**Comment by alex-s-gardner** [comment]
> Docs are now live

These docs are so impressive... what a great package!
